File: leetcode/Medium/0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseBetween(struct ListNode* head, int left, int right){
   struct ListNode* temp = (struct ListNode*)malloc(sizeof(struct ListNode));
    temp->val = -1;
    temp->next = NULL;
    struct ListNode* prev = temp;
    prev->next = head;
        for(int i=0;i<left-1;i++)
            prev = prev->next;
        struct ListNode* cur = prev->next;
        for(int i=0;i<right-left;i++){
            struct ListNode* ptr = prev->next;
            prev->next=cur->next;
            cur->next = cur->next->next;
            prev->next->next=ptr;
        }
        return temp->next;
}

File: leetcode/Medium/0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseBetween(
        self, head: Optional[ListNode], left: int, right: int
    ) -> Optional[ListNode]:
        temp = ListNode(-1)  # sentinel
        prev = temp
        temp.next = head
        for i in range(left - 1):
            prev = prev.next
        cur = prev.next
        for i in range(right - left):
            ptr = prev.next
            prev.next = cur.next
            cur.next = cur.next.next
            prev.next.next = ptr
        return temp.next
